src/main/java/marites/Marites.java

```diff
@@ -0,0 +1,312 @@
+    }
+
+    /**
+     * Parses a user's command, and executes it.
+     * @param userInput A line of the user's input.
+     * @return Output that must be passed back to the user.
+     */
+    private static String processUserCommand(String userInput) {
+        String[] commandSplit = splitCommandTypeAndCommand(userInput);
+        String commandOutput;
+        String commandType = commandSplit[0], command = commandSplit[1];
+        switch (commandType) {
+        case COMMAND_EXIT:
+            commandOutput = executeExit();
+            break;
+        case COMMAND_LIST:
+            commandOutput = executeListTasks();
+            break;
+        case COMMAND_MARK:
+            commandOutput = executeSetTaskStatus(command, true);
+            break;
+        case COMMAND_UNMARK:
+            commandOutput = executeSetTaskStatus(command, false);
+            break;
+        case COMMAND_ADD_TODO:
+        case COMMAND_ADD_DEADLINE:
+        case COMMAND_ADD_EVENT:
+            // All 3 cases fallthrough here
+            commandOutput = executeAddTask(commandType, command);
+            break;
+        case COMMAND_DELETE:
+            commandOutput = executeDelete(command);
+            break;
+        default:
+            commandOutput = UNKNOWN_COMMAND_MESSAGE;
+            break;
+        }
+        return commandOutput;
+    }
+
+    /*
+        The following methods take in a String, denoting a command.
+        They return a String, denoting the feedback after executing the command.
+     */
+
+    /**
+     * Executes a list command.
+     * @return The list of tasks.
+     */
+    private static String executeListTasks() {
+        StringBuilder taskList = new StringBuilder();
+        for (int i = 1; i <= tasks.size(); ++i) {
+            String task = String.format(LIST_TASK_ITEM_FORMAT_STRING, i, tasks.get(i-1));
+            taskList.append(task);
+        }
+        return taskList.toString();
+    }
+
+    /**
+     * Parses an add task command given by the user.
+     * @param taskType The task's type.
+     * @param command The user's command.
+     * @return A marites.task.Task object representing the task.
+     */
+    private static Task parseAddTask(String taskType, String command)
+        throws EmptyTaskDescriptionException, MissingParameterException, UnknownTaskTypeException {
+        String[] parametersSplit;
+        if (command.length() == 0) {
+            throw new EmptyTaskDescriptionException();
+        }
+        switch (taskType) {
+        case COMMAND_ADD_TODO:
+            return new Todo(command);
+        case COMMAND_ADD_DEADLINE:
+            parametersSplit = command.split(COMMAND_ADD_DEADLINE_TAG);
+            if (parametersSplit.length == 1) {
+                throw new MissingParameterException(COMMAND_ADD_DEADLINE_TAG);
+            }
+            return new Deadline(parametersSplit[0].strip(), parametersSplit[1].strip());
```

Comment:
You might want to consider extracting out each of the `case` block into a separate method for SLAP. Alternatively, you can also handle the exception elsewhere.